File: SDKs/tfl.gov.uk/v1/python/src/sdk/models/shared/tfl_api_presentation_entities_journeyplanner_itineraryresult.py
from dataclasses import dataclass, field
from typing import List,Optional
from dataclasses_json import dataclass_json
from sdk import utils
from . import *


@dataclass_json
@dataclass
class TflAPIPresentationEntitiesJourneyPlannerItineraryResult:
    r"""TflAPIPresentationEntitiesJourneyPlannerItineraryResult
    A DTO representing a list of possible journeys.
    """
    
    cycle_hire_docking_station_data: Optional[TflAPIPresentationEntitiesJourneyPlannerJourneyPlannerCycleHireDockingStationData] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('cycleHireDockingStationData') }})
    journey_vector: Optional[TflAPIPresentationEntitiesJourneyPlannerJourneyVector] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('journeyVector') }})
    journeys: Optional[List[TflAPIPresentationEntitiesJourneyPlannerJourney]] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('journeys') }})
    lines: Optional[List[TflAPIPresentationEntitiesLine]] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('lines') }})
    recommended_max_age_minutes: Optional[int] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('recommendedMaxAgeMinutes') }})
    search_criteria: Optional[TflAPIPresentationEntitiesJourneyPlannerSearchCriteria] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('searchCriteria') }})
    stop_messages: Optional[List[str]] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('stopMessages') }})
    

File: SDKs/tfl.gov.uk/v1/typescript/src/sdk/models/shared/tflapipresentationentitiesjourneyplanneritineraryresult.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { TflApiPresentationEntitiesJourneyPlannerJourneyPlannerCycleHireDockingStationData } from "./tflapipresentationentitiesjourneyplannerjourneyplannercyclehiredockingstationdata";
import { TflApiPresentationEntitiesJourneyPlannerJourneyVector } from "./tflapipresentationentitiesjourneyplannerjourneyvector";
import { TflApiPresentationEntitiesJourneyPlannerJourney } from "./tflapipresentationentitiesjourneyplannerjourney";
import { TflApiPresentationEntitiesLine } from "./tflapipresentationentitiesline";
import { TflApiPresentationEntitiesJourneyPlannerSearchCriteria } from "./tflapipresentationentitiesjourneyplannersearchcriteria";



// TflApiPresentationEntitiesJourneyPlannerItineraryResult
/** 
 * A DTO representing a list of possible journeys.
**/
export class TflApiPresentationEntitiesJourneyPlannerItineraryResult extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=cycleHireDockingStationData" })
  cycleHireDockingStationData?: TflApiPresentationEntitiesJourneyPlannerJourneyPlannerCycleHireDockingStationData;

  @SpeakeasyMetadata({ data: "json, name=journeyVector" })
  journeyVector?: TflApiPresentationEntitiesJourneyPlannerJourneyVector;

  @SpeakeasyMetadata({ data: "json, name=journeys", elemType: TflApiPresentationEntitiesJourneyPlannerJourney })
  journeys?: TflApiPresentationEntitiesJourneyPlannerJourney[];

  @SpeakeasyMetadata({ data: "json, name=lines", elemType: TflApiPresentationEntitiesLine })
  lines?: TflApiPresentationEntitiesLine[];

  @SpeakeasyMetadata({ data: "json, name=recommendedMaxAgeMinutes" })
  recommendedMaxAgeMinutes?: number;

  @SpeakeasyMetadata({ data: "json, name=searchCriteria" })
  searchCriteria?: TflApiPresentationEntitiesJourneyPlannerSearchCriteria;

  @SpeakeasyMetadata({ data: "json, name=stopMessages" })
  stopMessages?: string[];
}
